src/UserSpaceInstrumentation/OrbitUserSpaceInstrumentation.cpp

```diff
@@ -89,6 +89,13 @@ LockFreeUserSpaceInstrumentationEventProducer& GetCaptureEventProducer() {
   return producer;
 }
 
+// Provide a thread local bool to keep track of whether the current thread is inside the payload we
+// injected. If that is the case we avoid further instrumentation.
+bool& GetIsInPayload() {
+  thread_local bool is_in_payload = false;
+  return is_in_payload;
+}
```

Comment:
Why did you go for a function? Couldn't this be a simple global variable?

Reply:
That would create variable for all the threads regardless of whether they need one. Not a huge difference but I decided to avoid that.